lib/library.rb

```diff
@@ -0,0 +1,49 @@
+require './lib/author'
+
+class Library
+  attr_reader :name, :authors
+
+  def initialize(name)
+    @name = name
+    @authors = []
+  end
+
+  def add_author(author)
+    @authors << author
+  end
+
+  def books
+    if @authors != []
```

Comment:
You shouldn't need this conditional. If you run a map on an empty array, it will just return an empty array. Otherwise, great job building this method dynamically.